zng/resolver/context.go

```diff
@@ -52,6 +52,7 @@ func (c *Context) Reset() {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	c.table = c.table[:zng.IdTypeDef]
+	c.lut = make(map[string]int)
```

Comment:
On reset, you need to clear the lookup table to start from a clean slate otherwise you will lookup type names that point to the old type IDs from the previously reset type context.  Golang does not have a good way to clear a map.  You either create a new one or walk through the table and delete all the entries.

Reply:
I updated the comment.

Reply:
(This was the bug fix described in the PR summary.)